tests/components/modbus/test_modbus.py

```diff
@@ -17,56 +31,512 @@
+                    CONF_HOST: "modbusTestHost",
+                    CONF_PORT: 5501,
+                    CONF_NAME: "modbusTest",
+                    CONF_TIMEOUT: 30,
+                    CONF_DELAY: 10,
+                }
+            ]
+        },
+        {
+            DOMAIN: [
+                {
+                    CONF_TYPE: "serial",
+                    CONF_BAUDRATE: 9600,
+                    CONF_BYTESIZE: 8,
+                    CONF_METHOD: "rtu",
+                    CONF_PORT: "usb01",
+                    CONF_PARITY: "E",
+                    CONF_STOPBITS: 1,
+                }
+            ]
+        },
+        {
+            DOMAIN: [
+                {
+                    CONF_TYPE: "serial",
+                    CONF_BAUDRATE: 9600,
+                    CONF_BYTESIZE: 8,
+                    CONF_METHOD: "rtu",
+                    CONF_PORT: "usb01",
+                    CONF_PARITY: "E",
+                    CONF_STOPBITS: 1,
+                    CONF_NAME: "modbusTest",
+                    CONF_TIMEOUT: 30,
+                    CONF_DELAY: 10,
+                }
+            ]
         },
     ],
 )
+async def test_config_modbus(hass, do_config):
+    """Run test for modbus."""
+
+    await _config_helper(hass, do_config)
+    assert DOMAIN in hass.data
+    assert len(hass.data[DOMAIN]) == 1
+
+
 @pytest.mark.parametrize(
     "do_config",
     [
         {
-            CONF_TYPE: "tcp",
-            CONF_HOST: "modbusTestHost",
-            CONF_PORT: 5501,
-        },
-        {
-            CONF_TYPE: "serial",
-            CONF_BAUDRATE: 9600,
-            CONF_BYTESIZE: 8,
-            CONF_METHOD: "rtu",
-            CONF_PORT: "usb01",
-            CONF_PARITY: "E",
-            CONF_STOPBITS: 1,
+            DOMAIN: [
+                {
+                    CONF_TYPE: "tcp",
+                    CONF_HOST: "modbusTestHost",
+                    CONF_PORT: 5501,
+                    CONF_NAME: "modbusTest1",
+                },
+                {
+                    CONF_TYPE: "tcp",
+                    CONF_HOST: "modbusTestHost",
+                    CONF_PORT: 5501,
+                    CONF_NAME: "modbusTest2",
+                },
+                {
+                    CONF_TYPE: "serial",
+                    CONF_BAUDRATE: 9600,
+                    CONF_BYTESIZE: 8,
+                    CONF_METHOD: "rtu",
+                    CONF_PORT: "usb01",
+                    CONF_PARITY: "E",
+                    CONF_STOPBITS: 1,
+                    CONF_NAME: "modbusTest3",
+                },
+            ]
         },
     ],
 )
-async def test_config_modbus(hass, do_discovery, do_options, do_config):
-    """Run test for modbus."""
+async def test_config_multiple_modbus(hass, do_config):
+    """Run test for multiple modbus."""
+
+    await _config_helper(hass, do_config)
+    assert DOMAIN in hass.data
+    assert len(hass.data[DOMAIN]) == len(do_config[DOMAIN])
+
+
+def _config_entry():
```

Comment:
This function is named after a config entry but returns a hub. That is confusing.

You can define your own fixtures with pytest and inject them in your tests:

```python
import pytest

@pytest.fixture()
def modbus_hub():
    ...
    return hub

async def test_class_ModbusHub_logger(hass, caplog, modbus_hub):
    ...
```

Reply:
You really got me on the naming. I have added the pytest.fixture() which makes a lot of sense here.